tools/autograd/gen_variable_type.py

```diff
@@ -130,21 +130,33 @@
 RECORD_FUNCTION = CodeTemplate("""\
 profiler::RecordFunction profiler("${name}", Function::peek_at_next_sequence_nr());""")
 
+SELECT_OUTPLACE = CodeTemplate("""\
+  if (tracer_state->try_outplace) {
+    ${outplace_trace}
+  } else {
+    ${inplace_trace}
+  }
```

Comment:
Oh no is this effectively doubling the code size? Not sure if there's anything actionable but it's something to consider

Reply:
Yeah…I guess if we want to compile for binary size constrained environments we should probably strip the tracing stuff entirely though